src/Gitonomy/Git/Admin.php

```diff
@@ -168,6 +170,12 @@ public static function cloneRepository($path, $url, array $args = [], array $opt
 
     /**
      * This internal method is used to create a process object.
+     *
+     * @param string $command
+     * @param array  $args
+     * @param array  $options
+     *
+     * @return Process
```

Comment:
I think this is not the way to go. Instead we should add really PHP type.
It's easily doable for argument since PHP allow it in a BC manner. 
But the return type will be a huge BC break!

Reply:
@lyrixx Could you please elaborate on what you mean by BC?

You mean instead of 
```
     * @param string $command
     * @param array  $args
     * @param array  $options
     *
     * @return Process
     private static function getProcess($command, array $args = [], array $options = [])
```

one should write 

```
     * @param string $command
     * @param array  $args
     * @param array  $options
     private static function getProcess($command, array $args = [], array $options = []): Process
```

I'm all for it but I thought this wouldn't work with PHP 5.6. If it does work - it won't work with mixed types at least.

If this is ok, could I go one step further and add PHP typing for all the arguments as well to ditch the DocBlocks entirely?

Reply:
BC mean Backward Compatible. see https://symfony.com/bc ; We should use the same policy.

The patch would be:

```
    /**
     * This internal method is used to create a process object.
     *
     * @return Process
     */
    private static function getProcess(string $command, array $args = [], array $options = [])
```

Reply:
Unfortunately, this project is still compatible with PHP 5.6 😨 ! So wee cannot write this code (union). But with PHP 8+, you can 👍🏼 

I'll bump this and I'll do a new release (next week I guess)

Reply:
Perfect. Should I change everything that's not a union-type to php types instead of DocBlocks though like you required?

Reply:
yes, please :)

Reply:
I just figured out that [primitive type-hints weren't supported until PHP 7.0. ](https://stackoverflow.com/a/4103533/15139141)
So all I can do is add native type-hints only for arrays and objects.

Reply:
can you wait a bit? I'll bump some things...

Reply:
see #204 🎉